datatypes/mcs_decimal.h

```diff
@@ -73,21 +136,21 @@ constexpr int128_t Decimal128Null = int128_t(0x8000000000000000LL) << 64;
     @brief The function to produce scale multiplier/divisor for
     wide decimals.
 */
-inline void getScaleDivisor(int128_t& divisor, const int8_t scale)
+template<typename T>
+inline void getScaleDivisor(T& divisor, const int8_t scale)
 {
-    divisor = 1;
-    switch (scale/maxPowOf10)
+    if (scale < 0)
+    {
+        std::string msg = "getScaleDivisor called with negative scale: " + std::to_string(scale);
+        throw std::invalid_argument(msg);
+    }
+    if (scale < 19)
+    {
+        divisor = mcs_pow_10[scale];
+    }
+    else
     {
-        case 2:
-            divisor *= mcs_pow_10[maxPowOf10];
-            divisor *= mcs_pow_10[maxPowOf10];
-            break;
-        case 1:
-            divisor *= mcs_pow_10[maxPowOf10];
-        case 0:
-            divisor *= mcs_pow_10[scale%maxPowOf10];
-        default:
-            break;
+        divisor = mcs_pow_10_128[scale-18];
```

Comment:
Yes. And after some thought, I changed msc_pow-10 and mcs_pow_10_128 to signed, as it was in IDB_pow. Because getScaleDivisor is template based, it is most often a signed T. With the way it works, we still get a maximum scale of 38.